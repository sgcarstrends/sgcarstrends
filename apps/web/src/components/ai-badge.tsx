import { Chip } from "@heroui/chip";
import { Sparkles } from "lucide-react";

export const AIBadge = () => (
  <Chip color="primary" variant="bordered">
    <div className="flex items-center gap-1">
      <Sparkles className="size-4" />
      AI Analysis
    </div>
  </Chip>
);
