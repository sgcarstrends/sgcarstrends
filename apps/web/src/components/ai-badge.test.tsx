import { render } from "@testing-library/react";
import { AIBadge } from "./ai-badge";

describe("AIBadge", () => {
  it("should render correctly", () => {
    const { container } = render(<AIBadge />);
    expect(container).toMatchSnapshot();
  });

  it("should render sparkles icon", () => {
    render(<AIBadge />);

    // Check for the SVG element with the sparkles class
    const sparklesIcon = document.querySelector(".lucide-sparkles");
    expect(sparklesIcon).toBeVisible();
  });
});
