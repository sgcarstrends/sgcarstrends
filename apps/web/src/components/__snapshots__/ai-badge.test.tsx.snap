// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AIBadge > should render correctly 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap border-medium px-1 h-7 text-small rounded-full bg-transparent border-primary text-primary"
  >
    <span
      class="flex-1 text-inherit font-normal px-2"
    >
      <div
        class="flex items-center gap-1"
      >
        <svg
          class="lucide lucide-sparkles size-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.937 15.5A2 2 0 0 0 8.5 14.063l-6.135-1.582a.5.5 0 0 1 0-.962L8.5 9.936A2 2 0 0 0 9.937 8.5l1.582-6.135a.5.5 0 0 1 .963 0L14.063 8.5A2 2 0 0 0 15.5 9.937l6.135 1.581a.5.5 0 0 1 0 .964L15.5 14.063a2 2 0 0 0-1.437 1.437l-1.582 6.135a.5.5 0 0 1-.963 0z"
          />
          <path
            d="M20 3v4"
          />
          <path
            d="M22 5h-4"
          />
          <path
            d="M4 17v2"
          />
          <path
            d="M5 18H3"
          />
        </svg>
        AI Analysis
      </div>
    </span>
  </div>
</div>
`;
